path_oidc.go

```diff
@@ -371,6 +371,20 @@ func (b *jwtAuthBackend) authURL(ctx context.Context, req *logical.Request, d *f
 		return logical.ErrorResponse("role %q could not be found", roleName), nil
 	}
 
+	// If namespace will be passed around in state, don't store it in redirect_uri
+	namespace := ""
+	if config.PassNamespaceInState {
+		inputURI, err := url.Parse(redirectURI)
+		if err != nil {
+			return resp, nil
+		}
+		qParam := inputURI.Query()
+		namespace = qParam.Get("namespace")
```

Comment:
Should we error here if it does contain a namespace (as opposed to ignoring/removing it)?

Reply:
So in this block if `namespace` is a query parameter, it does need to be removed from the redirectURI, so it can be appended to `state` later on.

Though I did see that this was re-encoding the query parameters for redirectURI needlessly in the case where there are no namespaces to worry about. Updated in b0211022c9a857e6bb97844d9d9c4653e36b48bd.

Reply:
I see, thanks for the explanation. Is the namespace always going to be passed as a query parameter regardless of `NamespaceInState`? How is the namespace determined if it's coming from a request header or request path?

Reply:
So as things stand today with the redirect_uri, I think the namespace is always passed as a query parameter, since that's how the UI sends it to this auth_url endpoint. (The cli login for OIDC doesn't need to set a namespace parameter since it just opens a local listener.)

I did a little testing with the UI, and I don't see any headers coming through in `req.Headers` in the auth_url endpoint handler. I think it would be possible to infer the namespace from the mountpoint, or add another field to the auth_url API payload. Maybe that would be a cleaner way to specify the namespace? I'm not sure what else that would gain us, unless there's another UI-like system we're trying to integrate with.